parsec/core/gui/file_table.py

```diff
@@ -305,27 +324,29 @@ def item_clicked(self, row, column):
         except AttributeError:
             pass
 
-    def clear(self):
+    def clear(self) -> None:
         self.clearContents()
         self.setRowCount(0)
         self.previous_selection = []
 
-    def set_file_status(self, entry_id, synced=None, confined=None):
+    def set_file_status(
+        self, entry_id: EntryID, synced: bool = False, confined: bool = False
+    ) -> None:
         for i in range(1, self.rowCount()):
-            item = self.item(i, 0)
+            item = cast(IconTableItem, self.item(i, 0))
             if item and item.data(ENTRY_ID_DATA_INDEX) == entry_id.hex:
                 if (
                     item.data(TYPE_DATA_INDEX) == FileType.File
                     or item.data(TYPE_DATA_INDEX) == FileType.Folder
                 ):
                     if confined is not None:
-                        item.confined = confined
+                        item.is_confined = confined
                     if synced is not None:
                         item.is_synced = synced
                 return
 
-    def change_selection(self):
-        selected = self.selectedItems()
+    def change_selection(self) -> None:
+        selected = cast(list[IconTableItem], self.selectedItems())
```

Comment:
There are not all `IconTableItem`. They are `QTableWidgetItem` and should only be casted at the right time (so if the column is a `Column.ICON`, it can be cast as a `IconTableItem`)